hypothesis-python/RELEASE.rst

```diff
@@ -0,0 +1,3 @@
+RELEASE_TYPE: patch
+
+Fix failure to infer strategy for recursive types with default arguments.
```

Comment:
```suggestion
:func:`~hypothesis.strategies.from_type` now works in cases where we use
:func:`~hypothesis.strategies.builds` to create an instance and the constructor
has an argument which would lead to recursion.  Previously, this would raise
an error.

Thanks to Joachim B Haga for reporting and fixing this problem!
```

The credits line is optional - feel free to delete it if you prefer, or write in whatever name or handle you'd prefer it to be attributed to - but I do like saying "thanks" to contributors.  Also add yourself to the list in `AUTHORS.rst`!